pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -355,6 +355,14 @@ type StageCommitResult struct {
 	Commit *Commit `json:"stageCommit"`
 }
 
+type projectCreated struct {
+	Commit *Commit `json:"commit"`
+}
+
+type CreateProjectResult struct {
+	ProjectCreated *projectCreated `json:"createProject"`
+}
```

Comment:
This type needs to also embed the error types that the API can return. Once they are modeled, the buildplaner function for creating a project needs to inspect this result and handle the potential errors.

The commit type has some embedded errors which you might have to mirror here. https://github.com/ActiveState/cli/blob/master/pkg/platform/api/buildplanner/model/buildplan.go#L371

There are also some convenience functions that you can use and/or expand to handle the new error cases: https://github.com/ActiveState/cli/blob/master/pkg/platform/api/buildplanner/model/buildplan.go#L371

Finally, you will likely need a function to process the `ProjectCreatedError` specifically, like this one: https://github.com/ActiveState/cli/blob/master/pkg/platform/api/buildplanner/model/buildplan.go#L274

I realize this is a fair amount of information to take in and handle so let me know if I can help clear anything up.

Reply:
That's fine, but the `Type` and `Error` info needs to live in the `projectCreated` struct for some reason. If I put it in `CreateProjectResult`, I don't get any error information.

Reply:
Ahh okay, that has to do with how the response is structured. Our types need to reflect the JSON structure so we have to be careful with how it all comes together.